pkgs/applications/audio/fire/default.nix

```diff
@@ -0,0 +1,105 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, pkg-config
+, catch2
+, libX11
+, libXrandr
+, libXinerama
+, libXext
+, libXcursor
+, freetype
+, alsa-lib
+, Cocoa
+, WebKit
+, CoreServices
+, DiscRecording
+, CoreAudioKit
+}:
+
+let
+  # FetchContent replacement, check CMakeLists.txt for requested versions (Nixpkgs' Catch2 works)
+  readerwriterqueue = fetchFromGitHub {
+    owner = "cameron314";
+    repo = "readerwriterqueue";
+    rev = "v1.0.6";
+    sha256 = "sha256-g7NX7Ucl5GWw3u6TiUOITjhv7492ByTzACtWR0Ph2Jc=";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "fire";
+  version = "0.9.9";
+
+  src = fetchFromGitHub {
+    owner = "jerryuhoo";
+    repo = "Fire";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-HOXkoX5CjWFdWGP8ngzPgAJ3KKSZ10aKF3CIh4rGRpw=";
+  };
+
+  patches = [
+    ./0001-Remove-FetchContent-usage.patch
```

Comment:
Upstream is currently mixing Git submodules & CMake FetchContent, I've opened an issue asking them to decide on one (preferably submodules so we can just include them with `fetchSubmodules`): https://github.com/jerryuhoo/Fire/issues/46